packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -36,6 +36,22 @@ import PatternTransformationsMenu from './pattern-transformations-menu';
 export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 	const { replaceBlocks } = useDispatch( blockEditorStore );
 	const blockInformation = useBlockDisplayInformation( blocks[ 0 ].clientId );
+
+	// The following method is needed to allow a manual transform from the old to new
+	// gallery block formats. It can be removed once old galleries are automatically
+	// migrated on edit.
```

Comment:
Is it worth linking to this PR or adding a little detail about when the gallery is expected to be automatically migrated? Something to help prompt the removal of this code when appropriate.